app/assets/javascripts/components/generic_object/generic-object-definition-component.js

```diff
@@ -0,0 +1,234 @@
+  };
+
+  vm.addClicked = function() {
+    var saveMsg = sprintf(__('%s \"%s\" has been successfully added.'), vm.entity, vm.genericObjectDefinitionModel.name);
+    vm.saveWithAPI('post', '/api/generic_object_definitions/', vm.prepSaveObject(), saveMsg);
+  };
+
+  vm.prepSaveObject = function() {
+    vm.genericObjectDefinitionModel.properties.attributes = {};
+    vm.genericObjectDefinitionModel.properties.associations = {};
+    vm.genericObjectDefinitionModel.properties.methods = [];
+
+    vm.genericObjectDefinitionModel.properties.attributes = _.zipObject(
+      vm.genericObjectDefinitionModel.attribute_names,
+      vm.genericObjectDefinitionModel.attribute_types);
+
+    vm.genericObjectDefinitionModel.properties.associations = _.zipObject(
+      vm.genericObjectDefinitionModel.association_names,
+      vm.genericObjectDefinitionModel.association_classes);
+
+    vm.genericObjectDefinitionModel.properties.methods = vm.genericObjectDefinitionModel.method_names;
+
+    return {
+      name: vm.genericObjectDefinitionModel.name,
+      description: vm.genericObjectDefinitionModel.description,
+      properties: vm.genericObjectDefinitionModel.properties,
+    };
+  };
+
+  vm.saveWithAPI = function(method, url, saveObject, saveMsg) {
+    miqService.sparkleOn();
+    API[method](url, saveObject)
+      .then(miqService.redirectBack.bind(vm, saveMsg, 'success', vm.redirectUrl))
+      .catch(miqService.handleFailure);
+  };
+
+  vm.methodsArrayEmpty = function() {
+    var bArrayEmpty = true;
+    _.forEach(vm.genericObjectDefinitionModel.method_names, function(item) {
+      if (item !== '' && item !== null && item !== undefined) {
+        bArrayEmpty = false;
+        return false;
+      }
+      return true;
+    });
+    return bArrayEmpty;
+  };
+
+  vm.uniqueProperty = function(keyType) {
+    var primaryArray, secondaryArrays = [];
+    if (keyType === "Attributes") {
+      primaryArray = vm.genericObjectDefinitionModel.attribute_names;
+      secondaryArrays.push(vm.genericObjectDefinitionModel.association_names);
+      secondaryArrays.push(vm.genericObjectDefinitionModel.method_names);
+    }
+    if (keyType === "Associations") {
+      primaryArray = vm.genericObjectDefinitionModel.association_names;
+      secondaryArrays.push(vm.genericObjectDefinitionModel.attribute_names);
+      secondaryArrays.push(vm.genericObjectDefinitionModel.method_names);
+    }
+    if (keyType === "Methods") {
+      primaryArray = vm.genericObjectDefinitionModel.method_names;
+      secondaryArrays.push(vm.genericObjectDefinitionModel.attribute_names);
+      secondaryArrays.push(vm.genericObjectDefinitionModel.association_names);
+    }
+    return checkUniqueness(primaryArray, secondaryArrays);
+  };
+
+  // private functions
+  function getGenericObjectDefinitionFormData(response) {
+    Object.assign(vm.genericObjectDefinitionModel, response);
```

Comment:
question: shouldn't you assign the return value somewhere?  it looks like this line is noop?

Reply:
Despite the absence of an assignment variable, the `Object.assign` here seems to work, as in, it does assign the values in `response` to `vm.genericObjectDefinitionModel`.

But you are right, it does appear like a noop (even though it does the intended thing)
I will change it to -
```javascript
vm.genericObjectDefinitionModel = Object.assign(vm.genericObjectDefinitionModel, response);
```

(The `Object.assign` in other places does have a return value assigned to a variable.)

Reply:
thanks, I just re-read https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign and you are correct, not sure if you need to change anything...